### DISCUSSION TITLE
bug firefox translate

### DISCUSSION THREAD

**Discussion by agasthene** [mozilla/firefox-translations#519] [General]
Bonjour

L’extension ne fonctionne pas correctement a chaque fois qu’il demande pour traduire une page, chaque fois il y a « une  erreur qui s’est produite lors du chargement du moteur de traduction »

Sans préciser quel erreur. Peut importe si c’est avec la version standard ou la version développement.

cordialement

Hello

The extension does not work properly on firefox. Every time it asks to translate a page, every time there is "an error that occurred while loading the translation engine"

Without specifying which error. It doesn't matter if it is with the standard version or the development version.

What to do?

Sincerely

**Comment by kpu** [comment]
Version of Firefox, CPU model, and is private browsing on?

**Reply by agasthene** [reply]
It doesn't matter if it is with the **standard version or the development version** 
for the cpu, i don't understand the connection. for me there is none.
it is not the cpu that will make the browser not have access to the translation engine.
le mode privé est off.

**Reply by kpu** [reply]
In Help->About Firefox, what is the version number showing?  (Sorry I don't know the French menu titles)
Are you on a Mac M1 or M2? 
Is your CPU particularly old?  We don't support CPUs older than SSE4.1.

**Reply by agasthene** [reply]
If I didn't specify the version number, knowing the subtext, it's not the browser version that is at issue. There is no mention of the minimum processor and browser version needed to run this addon.

If the translation doesn't support older processors what about the browser? the question is irrelevant and lacks some thought. the pc is 10 years old

the more appropriate question would have been to know why the browser does not have access to the translation engine? how does it access it, what is its path. this is what needs investigation and answer before the version or old processor.

Translated with [www.DeepL.com/Translator](http://www.deepl.com/Translator) (free version)

**Reply by kpu** [reply]
https://addons.mozilla.org/en-US/firefox/addon/firefox-translations/

"A CPU that supports SSE4.1 extensions is required for this addon to function properly. If it doesn't, an error will be displayed when the translation is being started."

The reason is that Firefox does not support WebAssembly below SSE4.1. 

https://addons.mozilla.org/en-US/firefox/addon/firefox-translations/

"Une unité centrale qui prend en charge les extensions SSE4.1 est nécessaire pour cette addition à fonctionner correctement. Si ce n'est pas le cas, une erreur sera affichée lorsque le la traduction est en cours de démarrage."

La raison en est que Firefox ne prend pas en charge WebAssembly en dessous de SSE4.1.

Traduit avec le traducteur de bergamote.

**Reply by kpu** [reply]
I agree that the error message should be improved.  
Je suis d'accord pour dire que le message d'erreur devrait être amélioré. 

Traduit avec le traducteur de bergamote (there is only a free version).

**Reply by kpu** [reply]
https://bugzilla.mozilla.org/show_bug.cgi?id=1672911

**Reply by agasthene** [reply]
mmx(+) 3dnow(+) sse, se2, sse3, sse4a, x86-x64, amd-V

**Reply by kpu** [reply]
Try the "compat" release on https://translatelocally.com/ .  If that works, let me know and we'll go through how to get a working browser.

**Reply by agasthene** [reply]
si la version compat veut bien s'afficher il finit par crasher quand je colle un texte., qunad a la version 2019 ou 2022 elle présentent des problèmes sur le serveur. impossible a exécuter et le journal des événement de Windows rapporte des problème de lecture du fichier (corrompu) probablement venant du serveur .

if the compat version wants to display well it ends up crashing when i paste a text, qunad has the 2019 or 2022 version it present problems on the server. impossible to run and the windows event log reports problems reading the file (corrupted) probably coming from the server .

**Reply by agasthene** [reply]
f the compat version wants to display well it ends up crashing when i paste a text, qunad has the 2019 or 2022 version it present problems on the server. impossible to run and the windows event log reports problems reading the file (corrupted) probably coming from the server .

**Comment by marco-c** [comment]
We are not going to make any improvements or fixes to the addon since we are now focusing on the built-in version.
You can report problems with the built-in version on Bugzilla (https://bugzilla.mozilla.org/).
